tools/benchmark/src/Reporter.ts

```diff
@@ -193,8 +209,12 @@ export class BenchmarkReporter {
 			if (isResultError(value)) {
 				countFailure++;
 			} else {
-				benchmarkPeriodsSeconds.push(value.stats.arithmeticMean);
-				sumRuntime += value.elapsedSeconds;
+				if (value.stats !== undefined) {
+					benchmarkPeriodsSeconds.push(value.stats.arithmeticMean);
+				}
+				if (value.elapsedSeconds !== undefined) {
+					sumRuntime += value.elapsedSeconds;
+				}
```

Comment:
Nit, but also note that if one of the types in the `BenchmarkResult` union was just `BenchmarkData` instead of `Partial<BenchmarkData>`, the undefined check / coalescing would be unnecessary, and I think that's really the way we should go.
```suggestion
				sumRuntime += value.elapsedSeconds ?? 0;
```

Reply:
Hmm...then we have to keep the `MemoryBenchmarkStats` separate?